[Leb14] iP

Leb14 opened the pull request:
### YOURS
> Let me be YOURS.

Are you struggling with the due date? 
- [x] YES, I AM!
- [x] I NEED TIME MANAGEMENT TOOLS!
- [x] I WANT "YOURS"!

What YOURS can do for you:
- Add your tasks.
- Manage your tasks
- Be "YOURS"

And it is **FREE**!

What are you waiting for?
Download it from [here](https://github.com/nus-cs2103-AY2324S1/ip.git).

If you Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        Duke.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Leb14 committed:
Level - 0

Level - 0 increment

Leb14 committed:
Level - 1

Level - 1 increment

Leb14 committed:
Revert "Level - 1"

This reverts commit b1c09ef70da2ad43c697da3552b8a2e6a9af4dbb.

Leb14 committed:
Level - 1

Level - 1 increment

Leb14 committed:
Level - 3

Level 3 increment

Leb14 committed:
Revert "Level - 3"

This reverts commit 06e88a1b3e441cc3a06cbc622a14e222298ca248.

Leb14 committed:
Level - 2

Level - 2 increment

Leb14 committed:
Level - 3

Level - 3 increment

Leb14 committed:
Level - 4

Level 4 increment

Leb14 committed:
A - TextUiTesting

A - TextUiTesting

Leb14 committed:
Level - 5

Level 5 increment

Leb14 committed:
Level - 6

Level 6 increment

Leb14 committed:
Level - 7

Level - 7 increment

Leb14 committed:
Merge pull request #1 from Leb14/branch-level-7

Level - 7

Leb14 committed:
Level - 8

Level 8 increment

Leb14 committed:
Merge pull request #2 from Leb14/branch-level-8

Level - 8

Leb14 committed:
A-MoreOOP

A MoreOOP increment

Leb14 committed:
Merge pull request #3 from Leb14/add-gradle-support

Add gradle support

Leb14 committed:
Revert "Merge pull request #3 from Leb14/add-gradle-support"

This reverts commit 80a512c993c7814970891bc966c46519e9ce0945, reversing
changes made to 9f0cadd832d316939767e2c2a40e308e6bd06e18.

Leb14 committed:
A-packages

A packages increment

Leb14 committed:
Revert "A-packages"

This reverts commit ae31ae6cdd6c822918b47ad533a935c688196ed6.

Leb14 committed:
A-Packages

A-Packages increment

Leb14 committed:
Merge pull request #4 from Leb14/master

add gradle support

Leb14 committed:
Revert "Merge pull request #4 from Leb14/master"

This reverts commit dd21484746ae78104e97da0ee43e9e0d3b3ecf57, reversing
changes made to ed6d4d25bd4e34461919ffe785a6f10f4c087552.

Leb14 committed:
Revert "Merge pull request #4 from Leb14/master"

This reverts commit dd21484746ae78104e97da0ee43e9e0d3b3ecf57, reversing
changes made to ed6d4d25bd4e34461919ffe785a6f10f4c087552.

Leb14 committed:
Merge pull request #5 from Leb14/add-gradle-support

Add gradle support

Leb14 committed:
Revert "Merge pull request #5 from Leb14/add-gradle-support"

This reverts commit 5c822510e0c6685130c0dea9cf65d188caaecee9, reversing
changes made to de6046304635c0b88b590f86978760580b109df7.

Leb14 committed:
Revert "Merge pull request #5 from Leb14/add-gradle-support"

This reverts commit 5c822510e0c6685130c0dea9cf65d188caaecee9, reversing
changes made to de6046304635c0b88b590f86978760580b109df7.

Commit:
Merge branch 'master' of https://github.com/Leb14/ip

Leb14 committed:
Merge branch 'add-gradle-support'

Leb14 committed:
A-JUnit

A-Junit increment

Leb14 committed:
Amend FileHandler

Amend FileHandler

Leb14 committed:
Add Java Doc and Enhance Parser for Index Handling

add Java Doc

Enhance Parser to handle index numbers for delete, mark, and unmark commands to improve code structure and maintainability.

Leb14 committed:
Merge pull request #7 from Leb14/branch-A-JavaDoc

Add Java Doc and Enhance Parser for Index Handling

Leb14 committed:
Add Coding Standard and Fix some bugs

Leb14 committed:
Merge pull request #8 from Leb14/branch-A-CodingStandard

Add Coding Standard and Fix some bugs

Leb14 committed:
Add task search functionality

Leb14 committed:
Merge pull request #9 from Leb14/branch-level-9

Add task search functionality

chonguschonguschongus commented on src/main/java/Ui/Ui.java:
```diff
@@ -0,0 +1,198 @@
+                        + "  \\      /   | |    | |  |  |   |  |  |   ______|  \\ ___ \\\n"
+                        + "   |    |    | |    | |  |  |   |  |  |  \\  \\           | |\n"
+                        + "   |    |    | |____| |  |  |   |  |  |  | \\  \\     ____) |\n"
+                        + "   |____|    |________|  \\_________/  |__|   \\__\\  |_____/\n";
```
Wow! What a creative logo LeBBB 👍 😄 🤣

chonguschonguschongus commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,63 @@
+                System.out.println(e.getMessage());
+            }
+        }
+    }
```
Great implementation of OOP here. Looks fantastic 😸

chonguschonguschongus commented on src/main/java/TaskManager/Tasks.java:
```diff
@@ -0,0 +1,45 @@
+/**
+ * Abstract class for few types of tasks.
+ */
+public abstract class Tasks {
```
Class name should be singular. Same for the other childs of the Task class 😢

Leb14 replied:
Thanks for pointing out! I have refactored my class name and updated it in the recent commit.✅

chonguschonguschongus reviewed (commented):
Minor adjustment to class names aside, LGTM. Fantastic work Jin Hao 💌

Leb14 committed:
Rename Tasks, ToDos, Events and Deadlines class.

Tasks class to Task
ToDos class to ToDo
Events class to Event
Deadlines class to Deadline

JeremyYong128 commented on src/main/java/command/DeleteCommand.java:
```diff
@@ -0,0 +1,54 @@
+    }
+
+    /**
+     * Check whether the command is an exit command.
```
"Checks" would be better instead of "check"

Leb14 replied:
Thanks for pointing out! I have revised my JavaDoc and updated it in the recent commit.✅

JeremyYong128 commented on src/main/java/Storage/TaskList.java:
```diff
@@ -0,0 +1,127 @@
+     */
+    public TaskList findKeyword(String str) {
+        TaskList matchedList = new TaskList();
+        System.out.println("here");
```
This statement looks like it was used for debugging, can consider removing it

Leb14 replied:
Thanks for pointing out! I have removed the line in the recent commit.:white_check_mark:

JeremyYong128 reviewed (approved):
Nice and clean code, some minor areas for improvement identified but overall still very readable.

Leb14 committed:
Remove debugging line in TaskList class

Leb14 committed:
Refactor: Modify mark and unmark functions in Ui class, update JavaDoc

zekone commented on src/main/java/Parser/Parser.java:
```diff
@@ -0,0 +1,128 @@
+            return new MiscCommand();
+        }
+    }
+}
```
can consider using `switch-case` or `if-return` with line breaks in between instead of `else if` to improve readability, but still works good!

zekone commented on src/main/java/Storage/TaskList.java:
```diff
@@ -0,0 +1,126 @@
+     *
+     * @param i The index of the task to be removed.
+     */
+    public void remove(int i) {this.task.remove(i);}
```
can consider using egyptian brackets for consistency
```suggestion
    public void remove(int i) {
        this.task.remove(i);
    }
````

zekone commented on src/main/java/command/AddEventCommand.java:
```diff
@@ -0,0 +1,63 @@
+/**
+ * Command to add an Events task.
+ */
+public class AddEventCommand extends Command{
```
missing space (for all subclasses of `Command`, not just this)
```suggestion
public class AddEventCommand extends Command {
```

zekone commented on src/main/java/Parser/Parser.java:
```diff
@@ -0,0 +1,128 @@
+/**
+ * The `Parser` class is responsible for parsing user input commands into executable commands.
+ */
+public class Parser {
```
can consider renaming to `CommandParser` to be clearer.

zekone reviewed (commented):
Looks good to me! Clean and zero major problems.

Leb14 committed:
Checkstyle project

Leb14 committed:
Merge pull request #10 from Leb14/branch-A-CheckStyle

Checkstyle project

Leb14 committed:
Integrate GUI with main

Leb14 committed:
Merge pull request #11 from Leb14/branch-level-10

Integrate GUI with main

Leb14 committed:
Implement Assertion

 Add assertions to ensure task description and due date are not null in AddDeadlineCommand.
- Add assertions to ensure task description, start date, and end date are not null in AddEventCommand.
- Add assertions to ensure task description is not null in AddEventCommand.
- Add assertions to ensure errorMessage is not null in ErrorCommand.
- Add assertions to ensure keyword is not null in FindCommand.
-Refactor code in ListCommand to remove redundant if statement that has already been checked in t.toTaskStr()
- Add assertions to ensure index is not negative or zero in MarkCommand.
- Add assertions to ensure index is not negative or zero in UnmarkCommand.
- Refactor the code in Parser to improve code readability
- Refactor the code in FileHandler to improve code readability
- Add assertions to ensure various string field is not null in Deadline
- Add assertions to ensure various string field is not null in Todo
- Add assertions to ensure various string field is not null in Event

Leb14 committed:
Merge pull request #12 from Leb14/branch-A-Assertion

Add assertion

Leb14 committed:
Merge pull request #13 from Leb14/master

Updated code from master

Leb14 committed:
Improve Code Quality & add Javadocs

isMatch() is implemented in Event, Deadline and Todo class.

The implementation of this method is duplicated in these three classes.

To resolve this duplication:
1. Create a new package - taskutility
2. Create a new class - TaskMatcher
3. Implement the method in TaskMatcher.
4. Import the class to those classes to use the method.

Creating a new class to implement this method is a more effective approach.

Leb14 committed:
Refactor code in Parser class

The parseDeadlineCommand and parseEventCommand methods are overly verbose.

It make the code less readable.

Things that changed :
- Refactor the code that parse Date and Time to a new method
- Handle exception in the method as well

The code is now more readable and comprehensible.

Leb14 committed:
Rearrange the import & add Javadocs

Leb14 committed:
Remove unused line in run method

Leb14 committed:
Merge pull request #14 from Leb14/branch-A-CodeQuality

Code quality

Leb14 committed:
Create gradle.yml

Leb14 committed:
Update gradle.yml

Leb14 committed:
Update gradle.yml

Leb14 committed:
Update gradle.yml

Leb14 committed:
Update gradle.yml

Leb14 committed:
Update gradle.yml

Leb14 committed:
Merge pull request #16 from Leb14/branch-A-CI

Branch a CI

Leb14 committed:
Add EditCommand

Leb14 committed:
Merge pull request #18 from Leb14/branch-C-Update

Add EditCommand

Leb14 committed:
Enhance GUI

Leb14 committed:
Merge pull request #19 from Leb14/branch-A-BetterGui

Enhance GUI

Leb14 committed:
Refine GUI & Change Gradle  mainclass

Leb14 committed:
Merge pull request #20 from Leb14/branch-A-BetterGui

Refine GUI & Change Gradle  mainclass

Leb14 committed:
Update README.md

Leb14 committed:
Update README.md

Leb14 committed:
Handle past date exception & refine some parsing issue

Leb14 committed:
Merge branch 'master' of https://github.com/Leb14/ip

Leb14 committed:
Enhance Parser to handle more exception

Leb14 committed:
Update README.md

Leb14 committed:
Remove isExit for all the Command classes

Leb14 committed:
Merge branch 'master' of https://github.com/Leb14/ip

Leb14 committed:
Update README.md

Leb14 committed:
Add images to GFMD

Leb14 committed:
Merge branch 'master' of https://github.com/Leb14/ip

Leb14 committed:
Update README.md

Leb14 committed:
Add images

Leb14 committed:
Merge branch 'master' of https://github.com/Leb14/ip

Leb14 committed:
Add title to window & Add screenshot

Leb14 committed:
Update README.md

Leb14 committed:
Add features' screenshot

Leb14 committed:
Merge branch 'master' of https://github.com/Leb14/ip

Leb14 committed:
Update todo.png

Leb14 committed:
Update README.md

Leb14 committed:
Update README.md

Leb14 committed:
Update README.md
